python/graphstorm/gconstruct/construct_payload_graph.py

```diff
@@ -0,0 +1,592 @@
+            }
+        ]
+    }]
+
+    The function is expected to gather all the feature transformation into the first
+    node/edge config definition and return. For the above example, it will be:
+    [{
+        "node_id_col": "id",
+        "node_type": "user",
+        "features":     [
+            {
+                "feature_col":  "feat_one"
+            },
+            {
+                "feature_col":  "feat_two"
+            }
+        ]
+    }]
+
+    Parameters:
+    ------------
+        gconstruct_conf_list: dict
+            GConstruct Config Dict for either node or edge
+        type_name: str
+            Node/Edge Type Name
+        structure_type: str
+            One of "Node" or "Edge"
+    Return:
+    -------
+        dict: merged gconstruct config
+    """
+    conf_list = []
+    if structure_type == "Node":
+        col_name = "node_type"
+    elif structure_type == "Edge":
+        col_name = "relation"
+    else:
+        raise ValueError("Expect structure_type be one of Node/Edge")
+    for conf in gconstruct_conf_list:
+        if conf[col_name] == type_name:
+            conf_list.append(conf)
+
+    # Features may be defined in multiple block for one node/edge type
+    if len(conf_list) >= 2:
+        # Collect all feature transformation block into the first block
+        collected_features = []
+        for conf in conf_list:
+            if "features" in conf:
+                for feature in conf["features"]:
+                    collected_features.append(feature)
+        conf_list[0]["features"] = collected_features
+        conf_list = [conf_list[0]]
+    # if there is only one node/edge type definition block,
+    # no need to merge feature transformation config.
+    elif len(conf_list) == 1:
+        pass
+    else:
+        raise ValueError(f"Expect one node/edge type definition for {type_name}")
+    return conf_list[0]
+
+
+def merge_payload_input(payload_input_list):
+    """Merge the payload input within the same node/edge type
+
+    There may be multiple node/edge definitions within one node/edge type. For example:
+
+    [{
+        "node_type":    "user",
+        "node_id":      "u1",
+        "features":     {
+            "feat": [feat_val1]
+        }
+    },
+    {
+        "node_type":    "user",
+        "node_id":      "u2",
+        "features":     {
+            "feat": [feat_val2]
+        }
+    },
+    ]
+
+    This function is expected to group all the blocks with the same node_type/edge_type
+    to fit in the gconstruct feature transformation input.
+    The above return should be like:
+
+    [{
+        "node_type":    "user",
+        "node_id":      ["u1", "u2"]
+        "features":     {
+            "feat": [[feat_val1], [feat_val2]]
+        }
+    }]
+    Parameters:
+    ------------
+        payload_input_list: list of dict
+            input payload
+    Return:
+    -------
+        dict: merged payload input
+    """
+    merged_data_temp = {}
```

Comment:
Let's add a few more comments to this function because it's quite complex.

Reply:
The example result is here:
`
[{
        "node_type":    "user",
        "node_id":      ["u1", "u2"]
        "features":     {
            "feat": [[feat_val1], [feat_val2]]
        }
    }]
`

We can copy paste it to here as well.

Reply:
I modified the doc string to explain the purpose, and expected input and output.